### DISCUSSION TITLE
per node UI

### DISCUSSION THREAD

**Discussion by XanderXen** [nxt-dev/nxt_editor#271] [Ideas]
When building a rig with NXT, for example, it would be nice if you could associate a QT UI with any given node. If a node was used for importing skinClusters, for example, it would be nice when you highlight the node used for that, if you could see a section with an interface with all things skinCluster related, where the interface can be organized however you want and filled with useful functions, such as a button for writing out the skinCluster(s) data that are to be imported at this phase in the build. A button for launching other tools like NGSkin tools could be added to the interface for this, or any other skinCluster related tools. Another node that just imports and applies settings for animation controls  (color, shape, draw style ect) could have the interface for applying these changes interactively associated with this node, as well as a save button to pick up those settings the next time you build. Whatever you want to include that makes that step useful (for editing, and writing out aspects associated with a given step associated with a node). What would it take to extend NXT to add this sort of functionality?

**Comment by ImLucasBrown** [comment]
Just wanted to say, we've seen this, we've talked about, however we don't have a great answer right now.
One idea I've been tossing around is a callback system that would allow users to write plugins against, still just a napkin sketch. In your case we'd expose something like a `after_node_selected` callback. That would then let you run arbitrary Python in the context of the node and editor's internal state.
Again, just ideas at this point, no work has been done towards this (yet).
